tasks/get_current_archive_list/get_current_archive_list.py

```diff
@@ -0,0 +1,532 @@
+    "LastModifiedDate": "last_update timestamptz",
+    "ETag": "etag text",
+    "StorageClass": "storage_class text",
+}
+
+
+def generate_temporary_s3_column_list(manifest_file_schema: str) -> str:
+    """
+    Creates a list of columns that matches the order of columns in the manifest.
+    Columns used by our database are given standardized names.
+    Args:
+        manifest_file_schema: An ordered CSV representing columns in the CSV the manifest points to.
+
+    Returns: A string representing SQL columns to create.
+        Columns required for import but unused by orca will be filled in with `junk` values.
+        For example, 'orca_archive_location text, key_path text, size_in_bytes bigint, last_update timestamptz, etag text, storage_class text, junk7 text, junk8 text, junk9 text, junk10 text, junk11 text, junk12 text, junk13 text, junk14 text'
+
+    """
+    column_index = 0
+    manifest_file_schema = manifest_file_schema.replace(" ", "")
+    columns_in_csv = manifest_file_schema.split(",")
+    columns_in_postgres = []
+    for column_in_csv in columns_in_csv:
+        column_index = column_index + 1
+        postgres_column_name = column_mappings.get(
+            column_in_csv, "junk" + str(column_index) + " text"
+        )
+        columns_in_postgres.append(postgres_column_name)
+    return ", ".join(columns_in_postgres)
+
+
+# https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_PostgreSQL.S3Import.html
+# To make runnable, run 'CREATE EXTENSION IF NOT EXISTS aws_s3 CASCADE;'
+def create_temporary_table_sql(temporary_s3_column_list: str) -> TextClause:
+    """
+    Creates a temporary table to store inventory data.
+    Args:
+        temporary_s3_column_list: The list of columns that need to be created to store csv data.
+            Be very careful to avoid injection.
+    """
+    return text(
+        f"""
+        CREATE TEMPORARY TABLE s3_import(
+            {temporary_s3_column_list}
+        )
+        """
+    )
+
+
+def trigger_csv_load_from_s3_sql() -> TextClause:
+    """
+    SQL for telling postgres where/how to copy in the s3 inventory data.
+    """
+    return text(
+        """
+        SELECT aws_s3.table_import_from_s3(
+            's3_import',
+            '',
+            '(format csv)',
+            :report_bucket_name,
+            :csv_key_path,
+            :report_bucket_region,
+            :s3_access_key,
+            :s3_secret_key,
+            ''
+        )
+        """
+    )
+
+
+def translate_s3_import_to_partitioned_data_sql(report_table_name: str) -> TextClause:
+    """
+    SQL for translating between the temporary table and Orca table.
+    """
+    return text(
+        f"""
+        INSERT INTO orca.{report_table_name} (job_id, orca_archive_location, key_path, etag, last_update, size_in_bytes, storage_class)
+            SELECT :job_id, orca_archive_location, key_path, etag, last_update, size_in_bytes, storage_class
+            FROM s3_import
```

Comment:
Per my previous comment on adding a delete flag column wondering if we should add a where clause `where delete flag = False`

Reply:
Since we don't know how to recognize a delete flag, I'm hesitant to say that something isn't a delete flag.

Reply:
Make sure to document the setup for the report as we will need to let others know how to properly set this up either via terraform or the UI.

Reply:
Good point. Updated doc.